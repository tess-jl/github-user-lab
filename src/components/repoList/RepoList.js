import React from 'react';
import { useSelector } from 'react-redux';
import Loading from '../loading/Loading'; 
import Repo from '../repo /Repo';
// import { fetchUserRepos } from '../../services/gitHubApi';
import { selectRepoListLoading, selectRepoList } from '../../selectors/repoListSelectors';

const RepoList = () => {
  const loading = useSelector(selectRepoListLoading);
  const userRepos = useSelector(selectRepoList);

  if(loading) return <Loading />;

  const reposFromUser = userRepos.map(repo => (
    <li key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <ul>
      {reposFromUser}
    </ul> 
  );
};

export default RepoList;

