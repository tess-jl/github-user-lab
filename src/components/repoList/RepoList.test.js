import React from 'react';
import { shallow } from 'enzyme';
import RepoList from './RepoList';
import { Provider } from 'react-redux';
import store from '../../store';

describe('RepoList component', () => {
  it('renders RepoList', () => {
    const wrapper = shallow(<Provider store={store}><RepoList /></Provider>).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
