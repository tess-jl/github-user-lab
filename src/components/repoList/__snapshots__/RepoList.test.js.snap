// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList component renders RepoList 1`] = `<RepoList />`;
